### PR TITLE
1.28.0 blog post

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
r? @steveklabnik 
cc @rust-lang/release 

I'd appreciate feedback -- I feel like there's something missing, it'd be nice to have at least one more feature I think.

### CODE REVIEW

File: _posts/2018-08-02-Rust-1.28.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    // This will allocate memory using the system allocator.
+    v.push(1);
+}
+```
+
+However, sometimes you want to define a custom allocator for a given application
+domain. This is also relatively easy to do by implementing the `GlobalAlloc`
+trait. You can read more about how to do this in the [documentation].
+
+[`GlobalAlloc`]: https://doc.rust-lang.org/1.28.0/std/alloc/trait.GlobalAlloc.html
```
Comment (steveklabnik) [inline _posts/2018-08-02-Rust-1.28.md]: this should link to stable not 1.28.0

File: _posts/2018-08-02-Rust-1.28.md [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1280-2018-08-02
+
+## What's in 1.28.0 stable
+
+### Global Allocators
+
+1.28.0 adds the `#[global_allocator]` attribute, which allows Rust programs to
+set their allocator, as well as define new allocators by implementing the
+[`GlobalAlloc`] trait.
```
Comment (steveklabnik) [inline _posts/2018-08-02-Rust-1.28.md]: We may want to talk a bit more about what allocators are?

File: _posts/2018-08-02-Rust-1.28.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+}
+```
+
+However, sometimes you want to define a custom allocator for a given application
+domain. This is also relatively easy to do by implementing the `GlobalAlloc`
+trait. You can read more about how to do this in the [documentation].
+
+[`GlobalAlloc`]: https://doc.rust-lang.org/1.28.0/std/alloc/trait.GlobalAlloc.html
+[documentation]: https://doc.rust-lang.org/1.28.0/std/alloc/trait.GlobalAlloc.html
+
```
Comment (steveklabnik) [inline _posts/2018-08-02-Rust-1.28.md]: This is missing the boilerplate "for more see the release notes"

Comment (steveklabnik) [inline _posts/2018-08-02-Rust-1.28.md]: You said this post was missing something; I would possibly include https://github.com/rust-lang/rust/pull/50610/ here

File: _posts/2018-08-02-Rust-1.28.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+### Library stabilizations
+
+Several new APIs were stabilized this release:
+
+- [`Iterator::step_by`]
```
Comment (steveklabnik) [inline _posts/2018-08-02-Rust-1.28.md]: we usually try to 

1. group similar apis in a single bullet
2. leave a short description of what the api does

writing this is the worst part of writing these blog posts :)

**Comment by Mark-Simulacrum** [comment]
Okay, updated.

File: _posts/2018-08-02-Rust-1.28.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  = note: argument names cannot start with an underscore
+```
+
+See the [detailed release notes][notes] for more.
+
+### Library stabilizations
+
+We've already mentioned the stabilization of the `GlobalAlloc` trait, but
+another important stabilization is the [`NonZero`] types. These are wrappers
+around the standard unsigned integer types: `u8`, `u16, `u32`, `u64`, `u128`,
```
Comment (cuviper) [inline _posts/2018-08-02-Rust-1.28.md]: Missing the closing quote on `` `u16 ``

File: _posts/2018-08-02-Rust-1.28.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+format!("{_foo}", _foo = 6usize);
+```
+
+Previously, the error message emitted here was relatively poor:
+
+```
+error: invalid format string: expected `'}'`, found `'_'`
+  |
+2 |     format!("{_foo}", _foo = 6usize);
+  |                     ^^^^^^^^
```
Comment (dtolnay) [inline _posts/2018-08-02-Rust-1.28.md]: The carets are under `"{_foo}"`. https://play.rust-lang.org/?gist=df6fd444250a1fa26df8ecb3b57017b0&version=stable

File: _posts/2018-08-02-Rust-1.28.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+### Global Allocators
+
+Allocators are the way that programs in Rust obtain memory from the system at
+runtime. Previously, Rust did not allow changing the way memory is obtained,
+which prevented some use cases. With 1.28.0, the `#[global_allocator]` attribute
+is now stable, which allows Rust programs to set their allocator to the system
+allocator, as well as define new allocators by implementing the [`GlobalAlloc`]
+trait.
+
+The standard library provides a handle to the system allocator, which can be
```
Comment (sgrif) [inline _posts/2018-08-02-Rust-1.28.md]: Do you think it's worth mentioning jemalloc at some point? It took me a minute to realize this isn't the default behavior

File: _posts/2018-08-02-Rust-1.28.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+### Library stabilizations
+
+We've already mentioned the stabilization of the `GlobalAlloc` trait, but
+another important stabilization is the [`NonZero`] types. These are wrappers
+around the standard unsigned integer types: `u8`, `u16`, `u32`, `u64`, `u128`,
+and `usize`.
+
+This allows for size-optimization, for example, `Option<u8>` is two bytes large,
+but `Option<NonZeroU8>` is just one byte large. This is especially useful when
```
[the remark this answers is not in the document]
Comment (cuviper) [reply _posts/2018-08-02-Rust-1.28.md]: No, they're separate types now, e.g.
https://doc.rust-lang.org/nightly/std/num/struct.NonZeroU8.html

But maybe that should be clarified in mentioning "the `NonZero` types".

[the remark this answers is not in the document]
Comment (sgrif) [reply _posts/2018-08-02-Rust-1.28.md]: The wording of the previous paragraph definitely made me think that `NonZero<T>` was still a thing.

**Comment by Mark-Simulacrum** [comment]
I've pushed up an update -- @sgrif, can you tell me if that helps clarify things?

**Comment by sgrif** [comment]
Yes, I think it's much more clear now

**Pull request merged**